tasks/managed/embargo-check/embargo-check.yaml

```diff
@@ -47,6 +47,7 @@ spec:
               key: token
       script: |
         #!/usr/bin/env bash
+        set -x
```

Comment:
this is already on line 100.. not really sure how this helps

Reply:
It doesn't help all that much, but it will print more things for example for this line:
```
NUM_ISSUES=$(jq -cr '.releaseNotes.issues.fixed | length' "${DATA_FILE}")
```

And if nothing else, it makes it more consistent - starting with debug from line 10 (or something) seems kind of weird.

Reply:
`NUM_ISSUES=$(jq -cr '.releaseNotes.issues.fixed | length' "${DATA_FILE}")` is literally the only valuable thing you gain. But if you think it is important to have that, line 84 should be removed

Reply:
Also, adding `set -x` here isn't really all that important, I can remove it if you prefer that (but my comment above is still valid). What's more important is that there was no `set -x` for the second step (which is where this error happened).

Reply:
> But if you think it is important to have that, line 84 should be removed

No, that's there for a different reason. See the comment above line 84:
```
            # We set +x before the curl call and then capture the exit code. That can lead to breaking out of the
            # current loop iteration or continuing on, so let's set -x at the beginning of each iteration
```

On line 98, it will continue the loop while debugging is disabled. Of course there are more ways this could be done, e.g. putting `set -x` on top of line 96. Let me know if I should do that.

Reply:
why would we want it on line 96? if `OUTPUT=$(curl-with-retry --retry 3 "${AUTH_ARGS[@]}" "${API_URL}")` fails, we echo an error. What is wrong with not having `set -x` set for that? and if we need it, the one on line 100 should be before it. With this PR, you set it before the loop and as first thing in the loop. That doesn't make sense

Reply:
> why would we want it on line 96? if OUTPUT=$(curl-with-retry --retry 3 "${AUTH_ARGS[@]}" "${API_URL}") fails, we echo an error. What is wrong with not having set -x set for that?

What I'm saying is that if that condition is true, we would be without debugging from that point on (on the next loop iteration). So one way or another, we want to avoid that. I said line 96 as an example (because usually it's the easiest way to do it - to just enable debugging again as soon as you can, once you're not printing anything secret anymore - if we want to start selecting where it makes sense and where not, we will end up with `set -x`/`set +x` on every other line). It could also be above line 98.

Reply:
> and if we need it, the one on line 100 should be before it.

No. That the earliest you can do it because the way you implemented it. If you do this:
```
            OUTPUT=$(curl-with-retry --retry 3 "${AUTH_ARGS[@]}" "${API_URL}")
            set -x
            if [ $? -ne 0 ] ; then
```
You will mess up the condition, because now you're checking if the `set` went fine.

Reply:
With your add on line 50 here (which I am fine with keeping), what command currently executes with `set -x` that would *not* if you removed the `set -x` on line 83/84?